### PR TITLE
Aida 92

### PR DESCRIPTION
**Pull request by bdpham**
Added filtering capability to data-table component

### CODE REVIEW

File: src/app/components/data-table/data-table.component.html [hunk 1]
```diff
@@ -78,7 +98,7 @@
         <div class="filter-reset" *ngFor="let filter of getCloseableFilters()">
             <span class="filter-label" [title]="getFilterTitle(filter.value)">{{ getFilterCloseText(filter.value) }}</span>
             <button mat-icon-button class="neon-icon-button-small" aria-label="Close" tooltip-position="below"
-                    [matTooltip]="getRemoveFilterTooltip(filter.value)" (click)="removeLocalFilterFromLocalAndNeon(filter, false, true)">
+                    [matTooltip]="getRemoveFilterTooltip(filter.value)" (click)="removeLocalFilterFromLocalAndNeon(filter, true, false)">
```
Comment (eaciv) [inline src/app/components/data-table/data-table.component.html:101]: Did you get a chance to test your global filter with the local filter? Do they both get applied?

File: src/app/components/data-table/data-table.component.spec.ts [hunk 2]
```diff
@@ -33,13 +33,29 @@ import { DatasetService } from '../../services/dataset.service';
...
 import { TranslationService } from '../../services/translation.service';
 import { VisualizationService } from '../../services/visualization.service';
+import { FilterMock } from '../../../testUtils/MockServices/FilterMock';
+import { By } from '@angular/platform-browser';
+import * as neon from 'neon-framework';
 
-describe('Component: DataTable', () => {
-    let component: DataTableComponent;
-    let fixture: ComponentFixture<DataTableComponent>;
+fdescribe('Component: DataTable', () => {
```
Comment (eaciv) [inline src/app/components/data-table/data-table.component.spec.ts]: Can't commit fdescribe

File: src/app/components/data-table/data-table.component.ts [hunk 3]
```diff
@@ -498,10 +520,7 @@ export class DataTableComponent extends BaseNeonComponent implements OnInit, OnD
 
     // Get filters and format for each call in HTML
     getCloseableFilters() {
-        let closeableFilters = this.filters.map((filter) => {
-            return filter.value;
-        });
-        return closeableFilters;
+        return this.filters;
```
Comment (eaciv) [inline src/app/components/data-table/data-table.component.ts:523]: Good change!

File: src/app/config/config.simple_aida.json [hunk 4]
```diff
@@ -67,6 +67,8 @@
 					"database": "aida",
```
Comment (eaciv) [inline src/app/config/config.simple_aida.json:67]: Do we want to put the AIDA (Verdi?) configuration into the dash codebase?

**Review by eaciv — COMMENTED** [review]
Changes look good. I just need the change to fdescribe and answers to a couple other comments. All tests passed!

File: src/app/config/config.simple_aida.json (hunk 4 above)
Comment (bdpham) [reply src/app/config/config.simple_aida.json:67]: Yeah probably a good idea. I'll change that.

File: src/app/components/data-table/data-table.component.spec.ts (hunk 2 above)
Comment (bdpham) [reply src/app/components/data-table/data-table.component.spec.ts]: Of course I forgot this.. Will change ASAP.

File: src/app/components/data-table/data-table.component.ts (hunk 3 above)
Comment (bdpham) [reply src/app/components/data-table/data-table.component.ts:523]: 😃

File: src/app/components/data-table/data-table.component.html (hunk 1 above)
Comment (bdpham) [reply src/app/components/data-table/data-table.component.html:101]: The FilterMock testutil class does not call onSuccess, therefore, the removeFilter (removes local filter) method is never called in tests. I can't test the local filter, however the global filter test passes. The removeFilter does, however, have its own unit test that passes.

Comment (bdpham) [reply src/app/components/data-table/data-table.component.html:101]: Come see me if that doesn't make any sense or answer your question.

**Review by eaciv — APPROVED** [review]

**Pull request merged**